File: ios/Runner/ListTileNativeAdFactory.swift
import Flutter
import google_mobile_ads

class ListTileNativeAdFactory: FLTNativeAdFactory {
    func createNativeAd(_ nativeAd: GADNativeAd, customOptions: [AnyHashable : Any]?) -> FLTNativeAdView? {
        // Загружаем макет из ListTileNativeAd.xib
        guard let nibView = Bundle.main.loadNibNamed("ListTileNativeAd", owner: nil, options: nil)?.first as? GADNativeAdView else {
            return nil
        }

        // 1. Заголовок (Headline)
        (nibView.headlineView as? UILabel)?.text = nativeAd.headline

        // 2. Описание (Body)
        if let body = nativeAd.body {
            (nibView.bodyView as? UILabel)?.text = body
            nibView.bodyView?.isHidden = false
        } else {
            nibView.bodyView?.isHidden = true
        }

        // 3. Иконка (Icon)
        if let icon = nativeAd.icon {
            (nibView.iconView as? UIImageView)?.image = icon.image
            nibView.iconView?.isHidden = false
        } else {
            nibView.iconView?.isHidden = true
        }

        // 4. Кнопка призыва к действию (Call to Action) — если нужна
        if let callToAction = nativeAd.callToAction, let ctaView = nibView.callToActionView as? UIButton {
            ctaView.setTitle(callToAction, for: .normal)
            nibView.callToActionView?.isHidden = false
        } else {
            nibView.callToActionView?.isHidden = true
        }

        // Передаем объект нативной рекламы в View
        nibView.nativeAd = nativeAd

        return FLTNativeAdView(adView: nibView)
    }
}

File: android/app/src/main/kotlin/com/korshiles_app/ListTileNativeAdFactory.kt
package com.korshiles_app

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.google.android.gms.ads.nativead.NativeAd
import com.google.android.gms.ads.nativead.NativeAdView
import io.flutter.plugins.googlemobileads.NativeAdFactory

class ListTileNativeAdFactory(private val context: Context) : NativeAdFactory {
    override fun createNativeAd(
        nativeAd: NativeAd,
        customOptions: MutableMap<String, Any>?
    ): NativeAdView {
        val adView = LayoutInflater.from(context)
            .inflate(R.layout.list_tile_native_ad, null) as NativeAdView

        // Привязываем вьюхи к AdMob
        adView.headlineView = adView.findViewById(R.id.ad_headline)
        adView.bodyView = adView.findViewById(R.id.ad_body)
        adView.iconView = adView.findViewById(R.id.ad_icon)
        adView.callToActionView = adView.findViewById(R.id.ad_call_to_action)

        // Заполняем данные
        (adView.headlineView as TextView).text = nativeAd.headline

        if (nativeAd.body != null) {
            (adView.bodyView as TextView).text = nativeAd.body
            adView.bodyView?.visibility = View.VISIBLE
        } else {
            adView.bodyView?.visibility = View.GONE
        }

        if (nativeAd.icon != null) {
            (adView.iconView as ImageView).setImageDrawable(nativeAd.icon?.drawable)
            adView.iconView?.visibility = View.VISIBLE
        } else {
            adView.iconView?.visibility = View.GONE
        }

        if (nativeAd.callToAction != null) {
            (adView.callToActionView as Button).text = nativeAd.callToAction
            adView.callToActionView?.visibility = View.VISIBLE
        } else {
            adView.callToActionView?.visibility = View.GONE
        }

        adView.setNativeAd(nativeAd)
        return adView
    }
}